packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -56,6 +56,10 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
 		return this.nonPersistentCache.snapshotPrefetchResultCache;
 	}
 
+	public get joinSessionCache() {
```

Comment:
If we have to expose this publicly on the factory anyway, would it make more sense to just make this the API that returns the `ISocketStorageDiscovery` or undefined and do away with the free function? I know we discussed that the function feels like a weird API for the factory, but to me this feels just as weird, so maybe avoid the extra indirection?

Reply:
Exposing cache is better than putting this api on factory as its usage is very specific. There could be more such things and it won't make sense to expose such specific api on factory while the cache is something that belongs to factory and could be used for other purpose too. So exposing cache makes much more sense to me than putting this api on factory surface.

Reply:
Just to be clear, while this is public method, it's usage (ideally) should be constrained to driver. I believe we have \@internal mark-up, if so, I'd use it on this API, to discourage any usage by hosts (who have access to a concrete class).
That's a bit beyond this PR, but we should consider not exposing these classes at all, instead expose flat factory function (that returns IDocumentServiceFactory)

Reply:
That said, I think I agree with Alejandro.

Reply:
If you both agree, then I can put this on the factory